docs/external_objectives.rst

```diff
@@ -0,0 +1,318 @@
+Using external codes
+--------------------
+
+The ``ExternalObjective`` class allows external codes to be used within a DESC
+optimization. This takes a user-supplied function that does not need not be JAX
+transformable, and wraps it with forward finite differences. This is most useful for
+cases when the function needs to call external codes that are written in another
+language besides Python, or cannot be rewritten with JAX for some other reason. If the
+user function can be made JAX transformable, it is recommended to use the
+``ObjectiveFromUser`` class instead.
+
+This guide begins with a simple example of how to use a custom function with the
+``ExternalObjective``. It then uses the ``TERPSICHORE`` objective as a more realistic
+example, which can be used as a template for wrapping other codes. Finally, an example
+script is shown for how to run the TERPSICHORE optimization with multiprocessing.
```

Comment:
add a note that one must obtain the TERPSICHORE source themselves and that DESC does not have the code.

Reply:
either add note here or later in this doc